refarch-integrations/refarch-dms-integration/refarch-dms-integration-core/src/main/java/de/muenchen/refarch/integration/dms/adapter/out/s3/S3Adapter.java

```diff
@@ -0,0 +1,148 @@
+        for (String path : filePaths) {
+            for (Map.Entry<String, Long> stringLongEntry : getFileSizeForPath(path).entrySet()) {
+                if (map.put(stringLongEntry.getKey(), stringLongEntry.getValue()) != null) {
+                    throw new IllegalStateException("Duplicate key");
+                }
+            }
+        }
+        return map;
+    }
+
+    private Map<String, Long> getFileSizeForPath(final String path) throws DocumentStorageException {
+        if (path.endsWith("/")) {
+            return getSizesInFolderRecursively(path);
+        } else {
+            return Map.of(path, getFileSize(path));
+        }
+    }
+
+    private Map<String, Long> getSizesInFolderRecursively(final String folderPath) throws DocumentStorageException {
+        try {
+            return Objects.requireNonNull(documentStorageFolderRepository
+                    .getAllFileSizesInFolderRecursively(folderPath));
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException | DocumentStorageClientErrorException e) {
+            throw new DocumentStorageException("Metadata of a folder could not be loaded from url: " + folderPath, e);
+        }
+    }
+
+    private long getFileSize(final String filePath) throws DocumentStorageException {
+        try {
+            return Objects.requireNonNull(documentStorageFileRepository
+                    .getFileSize(filePath));
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException | DocumentStorageClientErrorException e) {
+            throw new DocumentStorageException("Metadata of a folder could not be loaded from url: " + filePath, e);
+        }
+    }
+
+    private List<Content> getFilesFromFolder(final String folderPath) throws DocumentStorageException {
+        try {
+            final List<Content> contents = new ArrayList<>();
+            final Set<String> filepath;
+            filepath = documentStorageFolderRepository.getAllFilesInFolderRecursively(folderPath);
+            if (Objects.isNull(filepath)) throw new DocumentStorageException("An folder could not be loaded from url: " + folderPath);
+            for (String file : filepath) {
+                contents.add(getFile(file));
+            }
+            return contents;
+        } catch (final DocumentStorageException | DocumentStorageServerErrorException | DocumentStorageClientErrorException e) {
+            throw new DocumentStorageException("An folder could not be loaded from url: " + folderPath, e);
+        }
+    }
+
+    private Content getFile(final String filePath) throws DocumentStorageException {
+        try {
+            final byte[] bytes;
+            bytes = this.documentStorageFileRepository.getFile(filePath, 3);
+            final String mimeType = fileService.detectFileType(bytes);
+            final String filename = FilenameUtils.getBaseName(filePath);
+
+            // check if mimeType exists
```

Comment:
probably better to remove the comments, because in the next line it says `isSupported`. To say `if exists` is kinda confusing here. no need for a comment at all
```suggestion
```